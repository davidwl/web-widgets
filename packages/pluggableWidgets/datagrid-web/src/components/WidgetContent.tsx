import { InfiniteBodyProps, useInfiniteControl } from "@mendix/widget-plugin-grid/components/InfiniteBody";
import classNames from "classnames";
import { ReactElement, ReactNode, createElement } from "react";
import { StickySentinel } from "./StickySentinel";

type PickProps = "hasMoreItems" | "setPage" | "isInfinite";

export type WidgetContentProps = {
    className?: string;
    children?: ReactNode;
    style?: React.CSSProperties;
} & Pick<InfiniteBodyProps, PickProps>;

export function WidgetContent({
    children,
    className,
    hasMoreItems,
    isInfinite,
    style,
    setPage
}: WidgetContentProps): ReactElement {
    const [trackScrolling, bodySize, containerRef] = useInfiniteControl({
        hasMoreItems,
        isInfinite,
        setPage
    });

    return (
        <div
            className={classNames(
                "widget-datagrid-content",
                "sticky-table-container",
                { "infinite-loading": isInfinite },
                className
            )}
            ref={containerRef}
            onScroll={isInfinite ? trackScrolling : undefined}
            style={isInfinite && bodySize > 0 ? { ...style, maxHeight: bodySize } : style}
        >
            <StickySentinel />
            {children}
        </div>
    );
}
