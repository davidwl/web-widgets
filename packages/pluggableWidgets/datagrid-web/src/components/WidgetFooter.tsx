import { createElement, ReactNode, ReactElement } from "react";
import { PagingPositionEnum } from "../../typings/DatagridProps";

type WidgetFooterProps = {
    pagingPosition: PagingPositionEnum;
    pagination: ReactNode;
} & JSX.IntrinsicElements["div"];

export function WidgetFooter(props: WidgetFooterProps): ReactElement | null {
    const { pagingPosition, pagination, ...rest } = props;
    return (
        <div {...rest} className="widget-datagrid-footer table-footer">
            {(pagingPosition === "bottom" || pagingPosition === "both") && pagination}
        </div>
    );
}
