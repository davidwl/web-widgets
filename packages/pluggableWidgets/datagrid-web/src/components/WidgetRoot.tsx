import { GridSelectionMethod } from "@mendix/widget-plugin-grid/selection/useGridSelectionProps";
import classNames from "classnames";
import { ReactElement, createElement } from "react";

type P = JSX.IntrinsicElements["div"];

export interface WidgetRootProps extends P {
    className?: string;
    selection?: boolean;
    selectionMethod: GridSelectionMethod;
    exporting?: boolean;
}

export function WidgetRoot(props: WidgetRootProps): ReactElement {
    const { className, selectionMethod, selection, exporting, children, ...rest } = props;
    return (
        <div
            {...rest}
            className={classNames(className, "widget-datagrid", {
                "widget-datagrid-exporting": exporting,
                "widget-datagrid-selectable-rows": selection,
                "widget-datagrid-selection-method-checkbox": selection && selectionMethod === "checkbox",
                "widget-datagrid-selection-method-click": selection && selectionMethod === "rowClick"
            })}
        >
            {children}
        </div>
    );
}
