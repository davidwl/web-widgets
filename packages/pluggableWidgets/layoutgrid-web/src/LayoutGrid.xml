<?xml version="1.0" encoding="utf-8" ?>
<widget id="com.mendix.widget.web.layoutgrid.LayoutGrid" pluginWidget="true" offlineCapable="true" xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../../../../node_modules/mendix/custom_widget.xsd">
    <name>LayoutGrid</name>
    <description />
    <studioProCategory>Containers</studioProCategory>
    <studioCategory>Containers</studioCategory>
    <helpUrl>https://docs.mendix.com/appstore/widgets/LayoutGrid</helpUrl>
    <properties>
        <propertyGroup caption="General">
            <propertyGroup caption="General">
                <property key="content" type="widgets">
                    <caption>Content</caption>
                    <description />
                </property>
            </propertyGroup>

            <propertyGroup caption="Visibility">
                <systemProperty key="Visibility" />
            </propertyGroup>
            <propertyGroup caption="Common">
                <systemProperty key="Name" />
                <systemProperty key="TabIndex" />
            </propertyGroup>
        </propertyGroup>
    </properties>
</widget>
