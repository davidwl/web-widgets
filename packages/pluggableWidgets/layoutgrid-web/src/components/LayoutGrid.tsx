import { createElement, CSSProperties, PropsWithChildren, ReactElement } from "react";
import classNames from "classnames";

export interface LayoutGridProps {
    name?: string;
    className: string;
    style?: CSSProperties;
    tabIndex?: number;
}

export function LayoutGrid(props: PropsWithChildren<LayoutGridProps>): ReactElement {
    const { tabIndex, style, className, children } = props;

    return (
        <div tabIndex={tabIndex} style={style} className={classNames(className, "mx-layoutgrid")}>
            {children}
        </div>
    );
}
