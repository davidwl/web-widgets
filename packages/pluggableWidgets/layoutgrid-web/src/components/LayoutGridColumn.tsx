import { createElement, CSSProperties, PropsWithChildren, ReactElement } from "react";
import classNames from "classnames";
import { VerticalAlignmentEnum } from "../../typings/LayoutGridColumnProps";

export interface LayoutGridColumnProps {
    name?: string;
    className: string;
    style?: CSSProperties;
    tabIndex?: number;
    weight: number;
    tabletWeight: number;
    phoneWeight: number;
    verticalAlignment: VerticalAlignmentEnum;
}

export function LayoutGridColumn(props: PropsWithChildren<LayoutGridColumnProps>): ReactElement {
    const { tabIndex, style, className, weight, tabletWeight, phoneWeight, children } = props;

    return (
        <div
            tabIndex={tabIndex}
            style={style}
            className={classNames(
                className,
                "col col-sm col-md col-lg col-xl",
                {
                    [`col-md-${weight}`]: weight > 0 && weight <= 12
                },
                {
                    [`col-lg-${weight}`]: weight > 0 && weight <= 12
                },
                {
                    [`col-xl-${weight}`]: weight > 0 && weight <= 12
                },
                {
                    [`col-sm-${phoneWeight}`]: phoneWeight > 0 && phoneWeight <= 12
                },
                {
                    [`col-md-${tabletWeight}`]: tabletWeight > 0 && tabletWeight <= 12
                },
                {
                    [`col-lg-${tabletWeight}`]: tabletWeight > 0 && tabletWeight <= 12
                },
                {}
            )}
        >
            {children}
        </div>
    );
}
