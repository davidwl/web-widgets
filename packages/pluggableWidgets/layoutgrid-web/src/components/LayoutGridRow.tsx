import { createElement, CSSProperties, PropsWithChildren, ReactElement } from "react";
import classNames from "classnames";
import { VerticalAlignmentEnum } from "../../typings/LayoutGridRowProps";

export interface LayoutGridRowProps {
    name?: string;
    className: string;
    style?: CSSProperties;
    tabIndex?: number;
    verticalAlignment: VerticalAlignmentEnum;
    spacingBetweenColumns: boolean;
}

export function LayoutGridRow(props: PropsWithChildren<LayoutGridRowProps>): ReactElement {
    const { tabIndex, style, className, children } = props;

    return (
        <div tabIndex={tabIndex} style={style} className={classNames(className, "row")}>
            {children}
        </div>
    );
}
