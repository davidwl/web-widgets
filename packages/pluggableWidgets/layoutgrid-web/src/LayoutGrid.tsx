import { createElement, ReactElement } from "react";

import { LayoutGrid as LayoutgridComponent } from "./components/LayoutGrid";
import { LayoutGridContainerProps } from "../typings/LayoutGridProps";

export default function LayoutGrid(props: LayoutGridContainerProps): ReactElement {
    const { content, name, tabIndex, style, class: className } = props;

    return (
        <LayoutgridComponent name={name} tabIndex={tabIndex} style={style} className={className}>
            {content}
        </LayoutgridComponent>
    );
}
