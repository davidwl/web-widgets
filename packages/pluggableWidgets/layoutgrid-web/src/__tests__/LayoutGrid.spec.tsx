import { shallow } from "enzyme";
import { createElement, ReactNode } from "react";

import { LayoutGrid, LayoutGridProps } from "../components/LayoutGrid";

describe("LayoutGrid", () => {
    const defaultLayoutGridProps: LayoutGridProps = {
        className: "className",
        style: {},
        tabIndex: 0
    };

    const defaultChildren: ReactNode = <div></div>;

    it("renders children", () => {
        const layoutgrid = shallow(<LayoutGrid {...defaultLayoutGridProps}>{defaultChildren}</LayoutGrid>);

        expect(layoutgrid).toMatchSnapshot();
    });
});
