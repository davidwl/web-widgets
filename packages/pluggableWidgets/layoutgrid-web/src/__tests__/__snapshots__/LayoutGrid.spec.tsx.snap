// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`LayoutGrid renders children 1`] = `
<div
  className="className mx-layoutgrid"
  style={{}}
  tabIndex={0}
>
  <div />
</div>
`;
